draft-ietf-mls-protocol.md

```diff
@@ -2547,12 +2548,10 @@ message at the same time, by taking the following steps:
   based on the proposals that are in the commit (see above), then it MUST be
   populated.  Otherwise, the sender MAY omit the `path` field at its discretion.
 
-* If populating the `path` field: Create an UpdatePath using the new tree. Any
-  new member (from an add proposal) MUST be exluded from the resolution during
-  the computation of the UpdatePath. The GroupContext for this operation uses
-  the `group_id`, `epoch`, `tree_hash`, and `confirmed_transcript_hash` values
-  in the initial GroupContext object.  The `leaf_key_package` for this
-  UpdatePath must have a `parent_hash` extension.
+* If populating the `path` field: Create an UpdatePath using the new tree and
+  the provisional GroupContext. Any new member (from an add proposal) MUST be
```

Comment:
You say use the provisional GroupContext, but the original text says to use the original GroupContext

Reply:
The "A member of the group applies a Commit message by taking the following steps:" section says

> * If the `path` value is populated: Process the `path` value using the new ratchet tree and the provisional GroupContext, to update the ratchet tree and generate the `commit_secret`:

So this change is to make the two sections consistent

Reply:
I agree it looks a bit inconsistent here indeed. I think this part is however the correct one and what you quote is confusing and should be changed instead.
The reason for my thinking is that the provisional group context is not known at this point in time and therefore the existing one should be used.

Reply:
If I'm reading the mlspp sources right, it looks like it uses the provisional GroupContext for this: https://github.com/cisco/mlspp/blob/main/src/state.cpp#L347-L359

> The reason for my thinking is that the provisional group context is not known at this point in time and therefore the existing one should be used.

The provisional group context is created two steps before this step.

I personally don't care which one is used, but they should be consistent.  For now, I'm going to follow what mlspp does, but if the consensus is that the original text here is correct, then I'm happy to change it.

Reply:
I think "provisional" is the right answer here.  The idea of the GroupContext used with TreeKEM is to capture the tree to which the secrets are encapsulated.  That tree is the one in the provisional GroupContext, to which the proposals have been applied.